jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/profiles/ProfileCredentialProviderFactory.kt

```diff
@@ -337,6 +339,13 @@ class ProfileCredentialProviderFactory : CredentialProviderFactory {
         .any { it.propertyExists(SSO_URL) }
 }
 
+private fun Profile.toCredentialType(): CredentialType? = when {
+    this.propertyExists(ProfileProperty.AWS_SESSION_TOKEN) -> CredentialType.StaticProfile
```

Comment:
while we're doing this, do we want to differentiate between session-keys (which should be a low number) and static-keys?

Reply:
Ya, probably. Will need to change it in the upstream

Reply:
https://github.com/aws/aws-toolkit-common/pull/82